packages/features/handleMarkNoShow.ts

```diff
@@ -20,97 +27,69 @@ const getResultPayload = async (attendees: { email: string; noShow: boolean }[])
   };
 };
 
-type ResponsePayload = {
-  attendees: { email: string; noShow: boolean }[];
-  noShowHost: boolean;
-  message: string;
-  messageKey: string | undefined;
-};
-
 const logFailedResults = (results: PromiseSettledResult<any>[]) => {
   const failed = results.filter((x) => x.status === "rejected") as PromiseRejectedResult[];
   if (failed.length < 1) return;
   const failedMessage = failed.map((r) => r.reason);
   console.error("Failed to update no-show status", failedMessage.join(","));
 };
 
-const handleMarkNoShow = async ({ bookingUid, attendees, noShowHost }: TNoShowInputSchema) => {
-  const responsePayload: ResponsePayload = {
-    attendees: [],
-    noShowHost: false,
-    message: "Failed to update no-show status",
-    messageKey: undefined,
-  };
-  try {
-    const attendeeEmails = attendees?.map((attendee) => attendee.email) || [];
-    if (attendees && attendeeEmails.length > 0) {
-      const allAttendees = await prisma.attendee.findMany({
-        where: {
-          AND: [
-            {
-              booking: {
-                uid: bookingUid,
-              },
-              email: {
-                in: attendeeEmails,
-              },
-            },
-          ],
-        },
-        select: {
-          id: true,
-          email: true,
-        },
-      });
+class ResponsePayloadWrapper {
+  attendees: { email: string; noShow: boolean }[];
+  noShowHost: boolean;
+  message: string;
+  messageKey: string | undefined;
 
-      const allAttendeesMap = allAttendees.reduce((acc, attendee) => {
-        acc[attendee.email] = attendee;
-        return acc;
-      }, {} as Record<string, { id: number; email: string }>);
-
-      const updatePromises = attendees.map((attendee) => {
-        const attendeeToUpdate = allAttendeesMap[attendee.email];
-        if (!attendeeToUpdate) return;
-        return prisma.attendee.update({
-          where: { id: attendeeToUpdate.id },
-          data: { noShow: attendee.noShow },
-        });
-      });
+  constructor() {
+    this.attendees = [];
+    this.noShowHost = false;
+    this.message = "Failed to update no-show status";
+    this.messageKey = undefined;
```

Comment:
I think we should set have just `messageKey` and `message` can be generated on the fly by `getPayload` having an argument t.